eclair-core/src/test/scala/fr/acinq/eclair/swap/SwapTransactionsSpec.scala

```diff
@@ -0,0 +1,107 @@
+import akka.testkit.TestProbe
+import fr.acinq.bitcoin.scalacompat.Crypto.PrivateKey
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, Crypto, Satoshi, SatoshiLong}
+import fr.acinq.eclair._
+import fr.acinq.eclair.blockchain.DummyOnChainWallet
+import fr.acinq.eclair.blockchain.OnChainWallet.MakeFundingTxResponse
+import fr.acinq.eclair.blockchain.bitcoind.BitcoindService
+import fr.acinq.eclair.blockchain.bitcoind.rpc.BitcoinCoreClient
+import fr.acinq.eclair.blockchain.fee.FeeratePerKw
+import fr.acinq.eclair.channel.publish.FinalTxPublisher
+import fr.acinq.eclair.channel.publish.TxPublisher.TxPublishContext
+import fr.acinq.eclair.swap.SwapTransactions._
+import fr.acinq.eclair.transactions.Transactions
+import fr.acinq.eclair.transactions.Transactions.{SwapClaimByCoopTx, SwapClaimByCsvTx, SwapClaimByInvoiceTx, checkSpendable}
+import grizzled.slf4j.Logging
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.AnyFunSuiteLike
+
+import java.util.UUID
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/**
+ * Created by remyers on 06/05/2022.
+ */
+
+class SwapTransactionsSpec extends TestKitBaseClass with AnyFunSuiteLike with BitcoindService with BeforeAndAfterAll with Logging {
+  val makerRefundPriv: PrivateKey = PrivateKey(randomBytes32())
+  val takerPaymentPriv: PrivateKey = PrivateKey(randomBytes32())
+  val paymentPreimage: ByteVector32 = randomBytes32()
+  val paymentHash: ByteVector32 = Crypto.sha256(paymentPreimage)
+  val amount: Satoshi = 30000 sat
+  val openingTxId: ByteVector32 = randomBytes32()
+  val openingTxOut: Int = 0
+  val claimInput: Transactions.InputInfo = makeSwapOpeningInputInfo(openingTxId, openingTxOut, amount, makerRefundPriv.publicKey, takerPaymentPriv.publicKey, paymentHash)
+
+  val csvDelay = 20
+  val localDustLimit: Satoshi = Satoshi(546)
+  val feeratePerKw: FeeratePerKw = FeeratePerKw(10000 sat)
+  val wallet = new DummyOnChainWallet()
+
+
+  override def beforeAll(): Unit = {
+    startBitcoind()
+    waitForBitcoindReady()
+  }
+
+  override def afterAll(): Unit = {
+    stopBitcoind()
+  }
+
+  def createFixture(): Fixture = {
+    val probe = TestProbe()
+    val watcher = TestProbe()
+    val bitcoinClient = new BitcoinCoreClient(bitcoinrpcclient)
+    val publisher = system.spawnAnonymous(FinalTxPublisher(TestConstants.Alice.nodeParams, bitcoinClient, watcher.ref.toTyped, TxPublishContext(UUID.randomUUID(), randomKey().publicKey, None)))
+    Fixture(bitcoinClient, publisher, watcher, probe)
+  }
+
+  case class Fixture(bitcoinClient: BitcoinCoreClient, publisher: ActorRef[FinalTxPublisher.Command], watcher: TestProbe, probe: TestProbe)
+
+  test("check validity of PeerSwap claim transactions") {
+    val f = createFixture()
+    import f._
+
+    val swapTxOut = makeSwapOpeningTxOut(amount, makerRefundPriv.publicKey, takerPaymentPriv.publicKey, paymentHash)
+    val future = wallet.makeFundingTx(swapTxOut.publicKeyScript, amount, feeratePerKw)
```

Comment:
can be simplified to 

```
wallet.makeFundingTx(swapTxOut.publicKeyScript, amount, feeratePerKw).pipeTo(probe.ref)
```

Reply:
much nicer! fixed in [3bd8c52](https://github.com/ACINQ/eclair/pull/2342/commits/3bd8c52b6a9989031aa9145b6868af49a2a3ccc9)